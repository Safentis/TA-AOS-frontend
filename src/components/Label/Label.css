.label {
  display: flex;
  flex-direction: column;
}

.label__content {
  color: var(--main-purple-color);
  cursor: pointer;
  font-size: 1.4rem;
  font-weight: 600;
  text-transform: uppercase;
  margin-bottom: 0.5rem;
}
