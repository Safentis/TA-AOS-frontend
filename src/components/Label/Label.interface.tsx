export interface Props {
  className?: string;
  content: string;
}
