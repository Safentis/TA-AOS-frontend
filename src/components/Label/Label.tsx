import React, { FC } from 'react';

import './Label.css';
import { Props } from './Label.interface';

const Label: FC<Props> = ({
  children,
  className = '',
  content,
}): React.ReactElement => {
  return (
    <label className={'label ' + className}>
      <span className="label__content">{content}</span>
      {children}
    </label>
  );
};

export default Label;
