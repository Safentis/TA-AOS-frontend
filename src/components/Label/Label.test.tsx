import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Label from './Label';

import { Props } from './Label.interface';

const setUp = (props: Props) => {
  return shallow(<Label {...props} />);
};

describe('<Label />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    className: 'custom-style',
    content: 'text-content',
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has 1 label element', () => {
      expect(component.find('label')).toBeTruthy();
    });

    it('has style "label" and "custom-style"', () => {
      expect(component.find('label').hasClass('label')).toBeTruthy();
      expect(
        component.find('label').hasClass('custom-style'),
      ).toBeTruthy();
    });

    it(`has content ${props.content}`, () => {
      expect(
        component.find('label').text() === props.content,
      ).toBeTruthy();
    });
  });
});

describe('<Label /> default props', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    content: 'mote of texts',
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has only one style "label"', () => {
      expect(component.find('label').hasClass('label')).toBeTruthy();
    });

    it(`has content ${props.content}`, () => {
      expect(
        component.find('label').text() === props.content,
      ).toBeTruthy();
    });
  });
});
