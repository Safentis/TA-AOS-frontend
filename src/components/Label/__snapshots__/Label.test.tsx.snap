// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> component does match to the snapshot 1`] = `
<label
  className="label custom-style"
>
  <span
    className="label__content"
  >
    text-content
  </span>
</label>
`;

exports[`<Label /> default props component does match to the snapshot 1`] = `
<label
  className="label "
>
  <span
    className="label__content"
  >
    mote of texts
  </span>
</label>
`;
